pkg/cmd/roachtest/tests/db_console_endpoints.go

```diff
@@ -0,0 +1,419 @@
+	if err != nil {
+		return errors.Wrap(err, "failed to verify response")
+	}
+
+	return nil
+}
+
+func initializeSchemaAndIDs(ctx context.Context, c cluster.Cluster, l *logger.Logger) error {
+	// Initialize some schema objects.
+	initTpcc := fmt.Sprint("./cockroach workload init tpcc --drop {pgurl:1}")
+	c.Run(ctx, option.WithNodes([]int{1}), initTpcc)
+
+	// Get SQL connection to query for a tableID, databaseID and fingerprintID.
+	conn, err := c.ConnE(ctx, l, 1)
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+
+	err = conn.QueryRowContext(ctx, "select id, \"parentID\" from system.namespace where name='warehouse'").Scan(&tableID, &databaseID)
+	if err != nil {
+		return err
+	}
+	l.Printf("Found tableID: %d, databaseID: %d", tableID, databaseID)
+
+	var fingerprintHexString string
+	err = conn.QueryRowContext(ctx, "select encode(fingerprint_id, 'hex') from crdb_internal.statement_statistics limit 1").Scan(&fingerprintHexString)
+	if err != nil {
+		return err
+	}
+	fingerprintID, err = strconv.ParseUint(fingerprintHexString, 16, 64)
+	if err != nil {
+		return err
+	}
+	l.Printf("Found fingerprintID hex %s, cast to int: %d", fingerprintHexString, fingerprintID)
+
+	return nil
+}
+
+func getEndpoints(t test.Test) []endpoint {
+	var endpoints []endpoint
+
+	endpoints = append(endpoints, parseEndpointsJSON(t, adminEndpointsJSON)...)
+	endpoints = append(endpoints, parseEndpointsJSON(t, apiV2EndpointsJSON)...)
+	endpoints = append(endpoints, parseEndpointsJSON(t, statusEndpointsJSON)...)
+
+	return endpoints
+}
+
+func parseEndpointsJSON(t test.Test, jsonStr string) []endpoint {
+	var endpoints endpointJSON
+	if err := json.Unmarshal([]byte(jsonStr), &endpoints); err != nil {
+		t.Fatal(errors.Wrap(err, "failed to parse endpoints JSON"))
+	}
+
+	var result []endpoint
+	for _, ep := range endpoints.Endpoints {
+		if ep.Skip {
+			t.L().Printf("skipping endpoint: %v", ep.URL)
+			continue
+		}
+		if ep.Verb == "POST" {
+			t.L().Printf("skipping POST endpoint: %v", ep.URL)
+			continue
+		}
+		result = append(result, endpoint{
+			url:    ep.URL,
+			method: HTTPMethod(ep.Verb),
+		})
+	}
+	return result
+}
+
+func (e endpoint) hasNodeID() bool {
+	return strings.Contains(e.url, nodeIDPlaceholder)
+}
+
+// getVerifyResponse returns the verifyResponse function to use for this endpoint.
+// If verifyResponse is not specified, returns defaultVerifyResponse.
+func (e endpoint) getVerifyResponse() func(resp *http.Response) error {
+	if e.verifyResponse == nil {
+		return defaultVerifyResponse
+	}
+	return *e.verifyResponse
+}
+
+// defaultVerifyResponse is the default implementation that just checks for 200 status code.
+func defaultVerifyResponse(resp *http.Response) error {
+	if resp.StatusCode != http.StatusOK {
+		body, _ := io.ReadAll(resp.Body)
+		return errors.Newf("unexpected status code: %d, body: %s", resp.StatusCode, body)
+	}
+	return nil
+}
+
+func fillPlaceholders(url string, nodeID string) (string, error) {
```

Comment:
This seems to be a good candidate for `text/template`:

example - https://go.dev/play/p/1Q80TaHDpkd
source - https://pkg.go.dev/text/template